### PR TITLE
Add the Devicetree compiler (`dtc`)

### PR DESCRIPTION
**Pull request by carlescufi**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/114001&drop=dogfoodAlpha

### CODE REVIEW

**Comment by carlescufi** [comment]
> @microsoft-github-policy-service agree [company="{your company}"]

@microsoft-github-policy-service agree [company="Nordic Semiconductor ASA"]

**Comment by carlescufi** [comment]
> @carlescufi please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Nordic Semiconductor ASA"

**Comment by carlescufi** [comment]
@Madhusudhan-MSFT @stephengillie @mdanish-kh 
This package includes an actual command-line application (`dtc.exe`) and installation and usage of it works fine for me locally.

File: manifests/d/Devicetree/dtc/1.5.1/Devicetree.dtc.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Devicetree.dtc
+PackageVersion: 1.5.1
+PackageLocale: en-US
+Publisher: Devicetree.org
```
Comment (stephengillie) [inline manifests/d/Devicetree/dtc/1.5.1/Devicetree.dtc.locale.en-US.yaml:7]:
```suggestion
Publisher: Devicetree
PublisherUrl: Devicetree.org
```

**Comment by mdanish-kh** [comment]
[Policy] zip-binary

@carlescufi We stopped accepting packages that suffer from the `.dll` issue since they're just a bad experience for most users and the actual fix hasn't been implemented.

**Comment by mdanish-kh** [comment]
[Policy] zip-binary

**Comment by beriberikix** [comment]
Hello! Been following this issue in part because I'm the one that asked help from @carlescufi to implement this 😅. It's unclear at least to me what the issue is - can someone help explain?

**Comment by mdanish-kh** [comment]
@beriberikix The package is not a standalone `.exe`. It depends on `.dlls` that although are included in the zip archive with the executable, WinGet does not have a consistent enough way of referencing them in every install scenario. As we cannot have a predictable and consistent behavior for every user installing the package, the repository is currently not accepting these packages termed as "Zipped-Binaries"

For more detail on how WinGet handles these zip installations and why these packages are blocked, please take a look at https://github.com/microsoft/winget-cli/discussions/3382#discussioncomment-6296017

**Comment by beriberikix** [comment]
Wasn't this suggested as the workaround to an issue with Winget? https://github.com/microsoft/winget-cli/issues/2711#issuecomment-1660638173

**Comment by mdanish-kh** [comment]
> Wasn't this suggested as the workaround to an issue with Winget? [microsoft/winget-cli#2711 (comment)](https://github.com/microsoft/winget-cli/issues/2711#issuecomment-1660638173)

Yes, installing in user shell with developer mode turned off would make the executable run properly, but even that is not ideal since it can easily pollute and fill up the PATH variable without the user noticing it. Since there is no right way currently to handle these packages, it was probably decided by the WinGet team to stop accepting new packages of this type until the issue is addressed properly in https://github.com/microsoft/winget-cli/issues/2711

**Comment by beriberikix** [comment]
Hello! I'm curious if we have a solution at this time.

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by Trenly** [comment]
@carlescufi - Support for zipped binaries has now been added and is available in the pipelines. This will require you to update your submission to ManifestVersion: 1.9.0 and set ArchiveBinariesDependOnPath: true in the installer node. To keep things clean, this PR is being closed. Please submit a new PR for the latest version of this package. Thank you for your contributions!

**Comment by carlescufi** [comment]
> @carlescufi - Support for zipped binaries has now been added and is available in the pipelines. This will require you to update your submission to ManifestVersion: 1.9.0 and set ArchiveBinariesDependOnPath: true in the installer node. To keep things clean, this PR is being closed. Please submit a new PR for the latest version of this package. Thank you for your contributions!

@Trenly thanks for the info! I am now working on this, do you have a suggestion on whether to start from scratch with `wingetcreate` or to modify the existing files in this  PR?

**Pull request closed without merging**